File: ios/Classes/style/ListScreenStyle.swift
//
//  ListScreenStyle.swift
//  exponea
//
//  Created by Adam Mihalik on 02/03/2023.
//

import Foundation
import ExponeaSDK

class ListScreenStyle {
    var emptyTitle: TextViewStyle?
    var emptyMessage: TextViewStyle?
    var errorTitle: TextViewStyle?
    var errorMessage: TextViewStyle?
    var progress: ProgressBarStyle?
    var list: AppInboxListViewStyle?

    init(
        emptyTitle: TextViewStyle? = nil,
        emptyMessage: TextViewStyle? = nil,
        errorTitle: TextViewStyle? = nil,
        errorMessage: TextViewStyle? = nil,
        progress: ProgressBarStyle? = nil,
        list: AppInboxListViewStyle? = nil
    ) {
        self.emptyTitle = emptyTitle
        self.emptyMessage = emptyMessage
        self.errorTitle = errorTitle
        self.errorMessage = errorMessage
        self.progress = progress
        self.list = list
    }

    func applyTo(_ target: AppInboxListViewController) {
        if let emptyTitleStyle = emptyTitle {
            emptyTitleStyle.applyTo(target.statusEmptyTitle)
        }
        if let emptyMessageStyle = emptyMessage {
            emptyMessageStyle.applyTo(target.statusEmptyMessage)
        }
        if let errorTitleStyle = errorTitle {
            errorTitleStyle.applyTo(target.statusErrorTitle)
        }
        if let errorMessageStyle = errorMessage {
            errorMessageStyle.applyTo(target.statusEmptyMessage)
        }
        if let progressStyle = progress {
            progressStyle.applyTo(target.statusProgress)
        }
        if let listStyle = list {
            listStyle.applyTo(target.tableView)
        }
    }
}

File: lib/src/data/model/list_screen_style.dart

import 'package:exponea/exponea.dart';

import '../util/Codable.dart';

class ListScreenStyle extends Encodable {
  final TextViewStyle? emptyTitle;
  final TextViewStyle? emptyMessage;
  final TextViewStyle? errorTitle;
  final TextViewStyle? errorMessage;
  final ProgressBarStyle? progress;
  final AppInboxListViewStyle? list;

  ListScreenStyle({this.emptyTitle, this.emptyMessage, this.errorTitle,
      this.errorMessage, this.progress, this.list});

  @override
  String toString() {
    return 'ListScreenStyle{emptyTitle: $emptyTitle, emptyMessage: $emptyMessage, errorTitle: $errorTitle, errorMessage: $errorMessage, progress: $progress, list: $list}';
  }

  @override
  Map<String, dynamic> encode() {
    return {
      'emptyTitle': emptyTitle?.encodeClean(),
      'emptyMessage': emptyMessage?.encodeClean(),
      'errorTitle': errorTitle?.encodeClean(),
      'errorMessage': errorMessage?.encodeClean(),
      'progress': progress?.encodeClean(),
      'list': list?.encodeClean()
    };
  }
}
